strawberryfields/api/connection.py

```diff
@@ -227,7 +227,15 @@ def get_job_result(self, job_id: str) -> Result:
             with io.BytesIO() as buf:
                 buf.write(response.content)
                 buf.seek(0)
+
                 samples = np.load(buf, allow_pickle=False)
+
+                if samples.dtype.type is np.int8:
```

Comment:
just wondering whether we shouldn't be more conservative here, and cast of the low-byte integer data type to int64, to avoid future issues (e.g. in case we'd switch to uint8 one day, or int16)?

```suggestion
                if samples.dtype in [np.int8, np.uint8, np.int16, np.uint16, np.int32, np.uint32]:
```

Reply:
How about the following?
```suggestion
                if np.issubdtype(samples.dtype, np.integer):
```

This is much more general, and will cast any integral type.